src/EndpointV2/EndpointV2Middleware.php

```diff
@@ -112,6 +117,9 @@ public function __invoke(CommandInterface $command)
     private function resolveArgs(array $commandArgs, Operation $operation) : array
     {
         $rulesetParams = $this->endpointProvider->getRuleset()->getParameters();
+
+        $this->resolveAccountIdArgument($commandArgs, $rulesetParams);
```

Comment:
Why not just do this conditionally based on whether or not `AccountId` and `AccountIdMode` are set?

Reply:
It is done conditionally at the method implementation. If AccountId is not part of the ruleset parameters or AccountIdEndpointMode is disabled then, it will abort the execution. Please see the method implementation.